### DISCUSSION TITLE
Run an effect when error message is not shown anymore

### DISCUSSION THREAD

**Discussion by danigb** [react-hook-form/react-hook-form#3219] [Q&A]
Hi, 

I have a problem. I want to detect when the validation error changed using an effect. Something like this:

```js
useEffect(() => {
  console.log('Errors changed')
}, [errors])
```

This is a running example: https://codesandbox.io/s/react-hook-form-get-started-forked-v6yzu

In the above example, when clicking "submit" the "This field is required" error is displayed and the effect is run. Everythink ok 👍  

The problem is when I start typing, the error disappears, but the effect is not run. In fact, the error is mutated (because the "errors object KEYS changed" message is in the console).

Am I missing something?

Thanks a lot for the library!

**Comment by bluebill1049** [comment]
put `formState` as dep

**Comment by danigb** [comment]
Hello, 

Thanks for the fast response! 🙏 

Although `formState` fires the effect, the problem is that is a new object in every render.

I my case, the `useEffect` changes the component state, so it triggers a new rendering. Because the `formState` is different, it triggers again the effect and at the end, creates an infinite loop.

To give more context: we're using a custom error object (implemented as an object state) to aggregate validation errors and other application errors. That approach worked well in previous versions of react-form, but when we tried to upgrade the library, it stop working. Although possible, changing the way we work with the custom error object is not an option (it would be a big change).

Any ideas? Thanks again

**Reply by bluebill1049** [reply]
Hmmm not sure why is that the case: take a look at my CSB: 
https://codesandbox.io/s/competent-cartwright-8fq8o?file=/src/App.tsx
